src/R/Components/Impl/InteractiveWorkflow/Commands/SessionInformationCommand.cs

```diff
@@ -35,12 +39,28 @@ public sealed class SessionInformationCommand : IAsyncCommand {
             return CommandResult.Executed;
         }
 
-        private void OnBrokerChanged(object sender, EventArgs e) {
+        private void OnRSessionInitialized(object sender, EventArgs e) {
             if (_interactiveWorkflow.RSession.IsRemote) {
-                PrintBrokerInformationAsync(reportTelemetry: true).DoNotWait();
+                ReplInitComplete().ContinueWith(async (t) => await PrintBrokerInformationAsync(reportTelemetry: true)).DoNotWait();
```

Comment:
I think right now all initialization code is in `RSession` itself.

The problem with initialization code living outside (if we do have it, or will ever have it) is that it's not clear how it's supposed to be ordered with respect to each other. Presumably some of that code depends on other initialization code running before. For example, for code that is inside `RSession` right now, it needs to load the `rtvs` package first, before it can start doing things like setting the working directory.

When it's all in one place inside the class itself, and everything outside of that class can only start using it once the init stuff is complete, then all is good - we can arrange the init code inside in the sequence that it needs to be in, and any outside user has the guarantee that they can rely on init. So in that world, there should be no need for a _public_ `RSession.Connected` event.